corehq/apps/users/validation.py

```diff
@@ -0,0 +1,52 @@
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+
+from corehq.apps.users.dbaccessors import user_exists
+from corehq.apps.users.exceptions import (
+    InvalidUsernameException,
+    ReservedUsernameException,
+    UsernameAlreadyExists,
+)
+from corehq.apps.users.util import format_username
+from corehq.apps.users.views.mobile import BAD_MOBILE_USERNAME_REGEX
+
+
+def validate_mobile_username(username, domain):
+    """
+    Returns the email formatted mobile username if valid
+    Otherwise raises custom exceptions based on the issue encountered with the username
+    :param username: str, required
+    :param domain: str, required
+    :return: str, email formatted mobile username
+    Example: validate_mobile_username('username', 'domain') -> 'username@domain.commcarehq.org'
+    """
+    if username is None:
```

Comment:
should this also check if the username is blank?

Reply:
That gets caught by the validation and is tested in `test_empty_username_raises_exception` which throws an `InvalidUsernameException`. If there are better reasons to put that check here, I'm open to it.

Reply:
OK that's fine. Just wanted to make sure it was covered.